src/Uno.Extensions.Core.Generators/PropertySelector/PropertySelectorCandidate.cs

```diff
@@ -1,43 +1,201 @@
+	private static string GenerateGetter(PropertySelectorPath path)
+		=> $"entity => entity{path.FullPath}"; // Note: this is expected to be the SimpleLambdaExpressionSyntax selector
+
+	private static string GenerateSetter(ITypeSymbol record, PropertySelectorPath path)
+	{
+		if (path.Parts.Count is 1)
+		{
+			var part = path.Parts[0];
+			var current = OrDefault(record) is { Length: > 0 } orDefault
+				? $"(current {orDefault})"
+				: "current";
+
+			return $"(current, updated_{part.Name}) => {current} with {{ {part.Name} = updated_{part.Name} }}";
+		}
+		else
+		{
+			var count = path.Parts.Count;
+			var type = record;
+			var current = new List<string>(count);
+			var updated = new List<string>(count);
+
+			for (var i = 1; i <= count; i++)
+			{
+				var part = path.Parts[i - 1];
+
+				if (i < count) // 'current_x' is a parameter of the delegate for the leaf element.
+				{
+					type = (type as INamedTypeSymbol)?.FindProperty(part.Name)?.Type;
+					if (type is null)
+					{
+						// The syntax is not valid (trying to get a property which does not exists), we cannot generate yet.
+						// But as anyway the compilation will fail, instead of failing the generator we just generate an exception.
+						return $"(_, __) => throw new InvalidOperationException(\"Cannot resolve property '{part.Name}' on '{type}'\")";
+					}
+
+					current.Add($"var current_{i} = current_{i - 1}{part.Accessor}{OrDefault(type)};");
+				}
+
+				updated.Add($"var updated_{i - 1} = current_{i - 1} with {{ {part.Name} = updated_{i} }};");
+			}
+
+			updated.Reverse();
+			return $@"(current_0, updated_{count}) =>
+			{{
+				{current.Align(4)}
+				{updated.Align(4)}
+
+				return updated_0;
+			}}".Align(0);
+		}
+
+		static string OrDefault(ITypeSymbol type)
+		{
+			if (type is { NullableAnnotation: NullableAnnotation.NotAnnotated })
+			{
+				// The type is annotated to NOT be nullable, we don't try to create a default instance.
+				return "";
+			}
+
+			if (type is not INamedTypeSymbol { IsRecord: true } record)
+			{
+				return "";
+			}
+
+			var ctor = record
+				.Constructors
+				.Where(ctor => ctor.IsAccessible() && !ctor.IsCloneCtor(record))
+				.OrderBy(ctor => ctor.HasAttribute<DefaultConstructorAttribute>() ? 0 : 1)
+				.ThenBy(ctor => ctor.Parameters.Length)
+				.Select(ctor => ctor.Parameters.Select(GetDefault).ToList())
+				.FirstOrDefault(ctorArgs => ctorArgs.All(arg => arg.hasDefault));
+
+			if (ctor is null)
+			{
+				return "";
+			}
+
+			return $" ?? new({ctor.Where(arg => arg.value is not null).Select(arg => arg.value).JoinBy(", ")})";
+		}
+
+		static (bool hasDefault, string? value) GetDefault(IParameterSymbol param)
+			=> param switch
+			{
+				{ HasExplicitDefaultValue: true } => (true, null),
+				{ Type.NullableAnnotation: NullableAnnotation.Annotated } => (true, $"default({param.Type})"),
+				{ } p when p.Type.IsNullable() => (true, $"default({param.Type})"),
+				_ => (false, null)
+			};
+	}
```

Comment:
I would move the generation code itself into a dedicated helper class. I would not expect to have such code in a `Candidate` class (e.g. keep it the `PropertySelectorsGenerationTool` but making it internal static)